src/Razor/src/Microsoft.AspNetCore.Razor.OmniSharpPlugin.StrongNamed/OmniSharpWorkspaceProjectStateChangeDetector.cs

```diff
@@ -92,10 +92,10 @@ private async Task InitializeSolutionAsync(Solution solution, CancellationToken
             // that abnormality.
             internal override void Workspace_WorkspaceChanged(object sender, WorkspaceChangeEventArgs args)
             {
-                _ = Workspace_WorkspaceChangedAsync(sender, args, CancellationToken.None);
+                _ = Workspace_WorkspaceChangedAsync(sender, args);
             }
 
-            private async Task Workspace_WorkspaceChangedAsync(object sender, WorkspaceChangeEventArgs args, CancellationToken cancellationToken)
```

Comment:
This is not reflective of this PR. But in general there seems to be a pattern of using no cancellationtoken sources even in some of the VS codepaths initializations. I am curious if there is a workspace token source that could be linked to be respectful of session close or missiong critical cancels and thereby a workspace dispose triggered cancel that we should be mindful of. In light of this being vscode, it makes it a bit more scary for me because I do not understand their multi threaded usage at all.

cc: @ryanbrandenburg @davidwengier

Reply:
Good call! We've discussed this in the past and agree that we should eventually be adding more cancellation token sources. Tracking issue is https://github.com/dotnet/razor-tooling/issues/4300

Reply:
Added that to deck for next sprint. I am surprised that task has escaped David Kean's eyes for an year now. :)